### ISSUE TITLE
Status of RDOS?

### ISSUE THREAD

**Issue by dara-ryerson** [RMS-open-standards/RDOS#23]
Hi people, 

I'm wondering if development on RDOS has stopped?  There hasn't been a commit for a year and a relatively simple pull request seems to be sitting there for over six months?

Are there any plans for the future?

Regards! Dara.

**Comment by cihanb** [comment]
Hi Dara, we have been taking in some feedback from internal and external teams and we'll be doing an update to the structure shortly. The new schema fixes a few issues and reflects the actual implementation in Risk Intelligence unified data store and we'd love your feedback once it is there. Were you interested in any particular part of RDOS?

**Comment by daragallagher** [comment]
Hi @cihanb, thanks for the reply.

I'm looking at aligning our internal models with the RDOS schema so I've been extracting the schema structure from this repository.  I've have some feedback/questions but am not sure where to ask.

One thing that bothers me but it's perhaps that I haven't studied it enough is that I would have been expecting more consistency in the naming of insurance contract concepts between the schema and CDL?  I see that there is CDL parsing code but I didn't find anything to link the parsed AST to the schema tables for example.

Another is a lack of an easy machine readable description of the schema (I'm writing a code generator).  I started by working with the json in rdos-validation-tools/rdos-schema/src/main/resources/ but found it lacked a lot of information provided in the Excel spreadsheet.  On the other hand, the Excel spreadsheet is missing some seed/reference data that I see in the SQL.  It worries me, for example, that I can't find any reference data for fields like StatusCode in insurance contract - are there other places where I should be looking?

I also have a vague fear because the model seems very oriented to property insurance - at least looking at the reference data.  I'm hoping to use it in a more general setting - in particular with other types of insurance.  Is is intended to be used for general insurance contracts?

I'm not sure this is the place to be posing questions like this - if there's somewhere else, let me know and I'll close this issue.

Regards! Dara.

**Comment by dara-ryerson** [comment]
Hi @cihanb, thanks for the reply.

I'm looking at aligning our internal models with the RDOS schema so I've been extracting the schema structure from this repository. I've have some feedback/questions but am not sure where to ask.

One thing that bothers me but it's perhaps that I haven't studied it enough is that I would have been expecting more consistency in the naming of insurance contract concepts between the schema and CDL? I see that there is CDL parsing code but I didn't find anything to link the parsed AST to the schema tables for example.

Another is a lack of an easy machine readable description of the schema (I'm writing a code generator). I started by working with the json in rdos-validation-tools/rdos-schema/src/main/resources/ but found it lacked a lot of information provided in the Excel spreadsheet. On the other hand, the Excel spreadsheet is missing some seed/reference data that I see in the SQL. It worries me, for example, that I can't find any reference data for fields like StatusCode in insurance contract - are there other places where I should be looking?

I also have a vague fear because the model seems very oriented to property insurance - at least looking at the reference data. I'm hoping to use it in a more general setting - in particular with other types of insurance. Is is intended to be used for general insurance contracts?

I'm not sure this is the place to be posing questions like this - if there's somewhere else, let me know and I'll close this issue.

Regards! Dara.

**Comment by cihanb** [comment]
thanks - certainly happy to help answer questions here. Starting in the machine readable part of the feedback, what format would be helpful for you?

**Comment by dara-ryerson** [comment]
Cool.

The actual format isn't as important as there being a single source for the schema structure information.  But since you've asked :) - I'd probably suggest expanding the idea of the Json format?  Json may not be perfect for this but at least it's easily machine readable while offering some human readability?

For example, I'd be inclined to add more structure than in the current data_type_validation.json file - instead of just providing a field name to type name, for each field you'd have a record per field containing not just its type but whether it's nullable, whether it's a primary or foreign key, it's valid values if any, default value, notes/comments, etc. 

And for each Json file/record representing a table, I'd include a list of seed/reference values.

**Comment by gajakannan** [comment]
Is RDOS still active?

**Comment by cihanb** [comment]
yes it is, however we have not been able to update it in a long while with all the changes the platform has gone through so it is quite out of date. We do plan to take another look at updates in 2024 but can't promise a specific date. the intelligent risk platform just delivered the capability to share exposure and loss data across tenant boundaries called "risk data exchange" in the platform. Plans right now is to move to share RDOS based data across customers using the platform and with that capability, we'd update the open standard. see it here: https://www.linkedin.com/posts/cihan-b_fantastic-leap-risk-data-exchange-is-now-activity-7141289201756512256-hlmN

**Comment by jbants** [comment]
Hi all,

We are looking at integrating RDOS with a specification and schema describing asset-level data for physical risk models derived from Global Climate models like CMIP6. 

We are following standards from the [open geospatial consortium (OGC)](https://ogcapi.ogc.org/processes/) to describe the location and using appropriate hazard indicators from organizations like [OS-C and their physical risk approach](https://os-climate.org/physical-risk-resilience/).

Are there any more plans to update RDOS?

**Comment by cihanb** [comment]
hi @jbants , RDOS is used as a standard storage system in the Intelligent Risk Platform (https://www.rms.com/risk-intelligence). However in the last few years, the implementation has evolved since our last update to RDOS. We do plan to update the implementation here in 2024/2025 but providing a firm timeline is hard. 
I am at Cihan.biyikoglu@moodys.com - Would be great to catch up on activities at OS-C and OGC to understand how we could contribute as Moodys. Could you reach out to see if we can find some time?
